### ISSUE TITLE
A simple program involving `BTreeSet` runs out of memory

### ISSUE THREAD

**Issue by zhassan-aws** [model-checking/kani#705]
I tried this code:

```rust
use std::collections::BTreeSet;

fn main() {
    let mut set = BTreeSet::<u32>::new();
    set.insert(0);
}
```

using the following command line invocation:

```
rmc test.rs --unwind 2
```

with RMC version: `8de5cd67609`

CBMC memory usage went up to 20GB after 1 minute and reached ~30 GB shortly after.

**Comment by zhassan-aws** [comment]
This got fixed by f088f9c97dc901e4fcd058d211d18f05bca6776f, in particular the switch to use `assert-false-assume-false` for undefined functions as opposed to just `assert-false`.

**Comment by tautschnig** [comment]
The changes in #1676 make an unwinding 2 terminate in 2.4 seconds (with the expected unwinding failure). An unwinding of 3 takes 3.1 seconds, no more unwinding failures.
